File: GameDatabase/AngularInterface/src/app/Models/Game.ts

export class Game {
  id: number;
  name: string;
  description: string;
  coverArtUrl: string;
  developerId: number;
  developerName: string;
  publisherId: number;
  publisherName: string;
  genreId: number;
  genre: string;
  platform: string;
  reviews: Review;
  releaseDate: Date;
    coverArtFile: any;
}

export class Review {
  authorId: string;
  gameId: number;
  title: string;
  text: string;
}

export class Search {
  Name: string;
}

File: GamesDatabaseBusinessLogic/Models/Game.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace GamesDatabaseBusinessLogic.Models
{
    public class Game : BaseEntity
    {
        public Game()
        {
            this.Reviews = new HashSet<Review>();
        }

        [Required]
        [StringLength(50, MinimumLength = 3)]
        public string Name { get; set; }

        [Required]
        public DateTime ReleaseDate { get; set; }

        [Required]
        public string CoverArtUrl { get; set; }

        public int DeveloperId { get; set; }
        [Required]
        public Developer Developer { get; set; }

        [Required]
        public string DeveloperName { get; set; }

        public int PublisherId { get; set; }
        [Required]
        public Publisher Publisher { get; set; }

        [Required]
        public string PublisherName { get; set; }

        public decimal GenreId { get; set; }
        [Required]
        public string Genre { get; set; }

        [Required]
        [StringLength(250, MinimumLength = 10)]
        public string Description { get; set; }

        [Required]
        public string Platform { get; set; }

        public bool IsFavouritedByUser { get; set; }

        public ICollection<Review> Reviews  { get; set; }

        public ICollection<User> Fans { get; set; }

        public ICollection<GamesFavourites> GamesFavourites { get; set; }
    }
}
